ballerina/observe_utils.bal

```diff
@@ -53,9 +53,13 @@ isolated function addTracingInfomation(TraceObserverContext|TraceInformation tra
     }
 }
 
-isolated function stopTracing(Context context) {
-    if tracingEnabled {
-        stopObserverContext(context);
+isolated function stopTracing(Context context, error? err = ()) {
+    if observabilityEnabled && tracingEnabled {
+        if err is () {
+            stopObserverContext(context);
+        } else {
+            stopObserverContextWithError(context, err);
+        }
     }
```

Comment:
```suggestion
    if !tracingEnabled {
        return;
    }
    if err is () {
        stopObserverContext(context);
        return;
    }
    stopObserverContextWithError(context, err);
```

Reply:
We have to check observabilityEnabled or not as well. Otherwise there will be NPE